skipchain/skipchain.go

```diff
@@ -975,27 +958,16 @@ func (s *Service) forwardLinkLevel0(src, dst *SkipBlock) error {
 		return errors.New("Wrong BFT-signature: " + err.Error())
 	}
 
-	if _, err := s.db.StoreBlocks([]*SkipBlock{src, dst}); err != nil {
-		return errors.New("couldn't store new forward link or new block: " + err.Error())
-	}
-	var proof []*SkipBlock
-	pointer := s.db.GetByID(dst.SkipChainID())
-	for {
-		if pointer == nil {
-			return errors.New("missing skipblock in proof creation")
-		}
-		proof = append(proof, pointer)
-		if pointer.Hash.Equal(dst.Hash) {
-			break
-		}
-		highest := len(pointer.ForwardLink) - 1
-		if highest < 0 {
-			return errors.New("proof chain has a missing forward link")
-		}
-		pointer = s.db.GetByID(pointer.ForwardLink[highest].To)
-	}
-	s.startPropagation(proof)
-	return nil
+	ro := roster.Concat(dst.Roster.List...)
+
+	// Propagate the forward link and the new block as the roster
+	// could have changed, meaning one node cannot know the block
+	return s.startForwardLinkPropagation(ro, &PropagateForwardLink{
+		ForwardLink: fwd,
+		Newest:      dst,
```

Comment:
Needed: For optimizing it correctly, you have to store the block in the `bftForwardLinkLevel0`, so this will not send it again.